test/test.ads.js

```diff
@@ -325,299 +268,160 @@ QUnit.test('changing src does not trigger "contentupdate" during ad playback', f
-  this.player.ads.endLinearAdMode();
-  assert.strictEqual(this.player.ads.state, 'content-resuming');
-  assert.strictEqual(endedSpy.callCount, 0, 'we should not have gotten an ended event yet');
+  assert.strictEqual(endedSpy.callCount, 0, 'ended was redispatched as contentended');
 
-  this.clock.tick(1000);
-  assert.strictEqual(endedSpy.callCount, 1, 'we should have fired ended from the timeout');
+  this.player.ads.startLinearAdMode(); // start postroll
+  this.player.ads.endLinearAdMode();
+  assert.strictEqual(endedSpy.callCount, 1, 'ended event happened');
 });
 
-QUnit.test('an "ended" event is not fired in "content-resuming" via a timeout if fired naturally', function(assert) {
+QUnit.test('ended events', function(assert) {
   var endedSpy = sinon.spy();
-
-  assert.expect(6);
-
   this.player.on('ended', endedSpy);
-  assert.strictEqual(this.player.ads.state, 'content-set');
 
+  this.player.trigger('loadstart');
   this.player.trigger('adsready');
-  assert.strictEqual(this.player.ads.state, 'ads-ready');
-
   this.player.trigger('play');
-  this.player.trigger('adtimeout');
-  this.player.trigger('ended');
-  assert.strictEqual(this.player.ads.state, 'postroll?');
+  this.player.trigger('adtimeout'); // Preroll times out
+  this.player.trigger('ended'); // Content ends (contentended)
 
-  this.player.ads.startLinearAdMode();
+  this.player.ads.startLinearAdMode(); // Postroll starts
   this.player.ads.endLinearAdMode();
-  assert.strictEqual(this.player.ads.state, 'content-resuming');
-  assert.strictEqual(endedSpy.callCount, 0, 'we should not have gotten an ended event yet');
+
+  assert.strictEqual(endedSpy.callCount, 1, 'ended event after postroll');
 
   this.player.trigger('ended');
-  assert.strictEqual(endedSpy.callCount, 1, 'we should have fired ended from the timeout');
+  assert.strictEqual(endedSpy.callCount, 2, 'ended event after ads done');
 });
 
-QUnit.test('adserror in ad-playback transitions to content-playback and triggers adend', function(assert) {
-  var spy;
+QUnit.test('adserror during ad playback triggers adend', function(assert) {
+  var adendSpy = sinon.spy();
 
-  assert.strictEqual(this.player.ads.state, 'content-set');
+  this.player.on('adend', adendSpy);
 
   this.player.trigger('adsready');
-
-  assert.strictEqual(this.player.ads.state, 'ads-ready');
-
   this.player.trigger('play');
   this.player.ads.startLinearAdMode();
-  spy = sinon.spy();
-  this.player.on('adend', spy);
-  this.player.trigger('adserror');
-  assert.strictEqual(this.player.ads.state, 'content-resuming');
-  assert.strictEqual(this.player.ads.triggerevent, 'adserror', 'The reason for content-resuming should have been adserror');
 
-  this.player.trigger('playing');
-  assert.strictEqual(this.player.ads.state, 'content-playback');
-  assert.strictEqual(spy.getCall(0).args[0].type, 'adend', 'adend should be fired when we enter content-playback from adserror');
+  assert.strictEqual(adendSpy.callCount, 0, 'no adend yet');
```

Comment:
what is this test testing now?

Reply:
Updated this test to do what it says it should